src/encoded/static/components/gene.js

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import * as globals from './globals';
+import { Breadcrumbs } from './navigation';
+import { ExperimentTable } from './dataset';
+import { DbxrefList, dbxrefHref } from './dbxref';
+import { RelatedItems } from './item';
+
+
+/* eslint-disable react/prefer-stateless-function */
+class Gene extends React.Component {
+    render() {
+        const context = this.props.context;
+        const itemClass = globals.itemClass(context, 'view-detail key-value');
+        let geneLink;
+        let geneRef;
+        let baseName;
+        let sep;
+
+        if (context.organism.name === 'human') {
+            geneLink = dbxrefHref('HGNC', context.gene_name);
+        } else if (context.organism.name === 'mouse') {
+            const mgiRef = _(context.dbxref).find(ref => ref.substr(0, 4) === 'MGI:');
+            if (mgiRef) {
+                geneLink = dbxrefHref('MGI', mgiRef);
+            }
+        } else if (context.organism.name === 'dmelanogaster' || context.organism.name === 'celegans') {
+            const organismPrefix = context.organism.name === 'dmelanogaster' ? 'FBgn' : 'WBGene';
+            const baseUrl = context.organism.name === 'dmelanogaster' ? globals.dbxrefPrefixMap.FlyBase : globals.dbxrefPrefixMap.WormBase;
+            geneRef = _.find(context.dbxref, ref => ref.indexOf(organismPrefix) !== -1);
+            if (geneRef) {
+                sep = geneRef.indexOf(':') + 1;
+                baseName = geneRef.substring(sep, geneRef.length);
+                geneLink = baseUrl + baseName;
+            }
+        }
```

Comment:
Unless I’m not understanding something, I think this whole if/else if/else if block can be eliminated by just using the `<DbxrefList>` component, and adding dependencies to the schema. The danger is that `dbxrefHref` isn’t complete; it doesn’t do any pre- or post-processing of the URL. They might not be needed in this case, but this seems like a dangerous way to go unless you really had a reason that `<DbxrefList>` wouldn’t work, in which case that should be commented here, otherwise everyone working on this code in the future will wonder why it duplicates so much work in dbxref.js. Also, the `geneLink` variable can end up undefined with a couple different paths through this block, and you’ll have an undefined `<a>` href later in this component if that happens.